playground/backend/cmd/remove_unused_snippets.go

```diff
@@ -16,44 +16,130 @@
 package main
 
 import (
+	"beam.apache.org/playground/backend/internal/logger"
 	"context"
+	"flag"
 	"fmt"
 	"os"
-	"strconv"
 	"time"
 
 	"beam.apache.org/playground/backend/internal/db/datastore"
 	"beam.apache.org/playground/backend/internal/db/mapper"
 )
 
-func main() {
-	if len(os.Args) < 3 {
-		fmt.Println("Go must have at least three arguments")
-		return
+func createDatastoreClient(ctx context.Context, projectId string) (*datastore.Datastore, error) {
+	pcMapper := mapper.NewPrecompiledObjectMapper()
+	db, err := datastore.New(ctx, pcMapper, projectId)
+	if err != nil {
+		fmt.Printf("Couldn't create the database client, err: %s \n", err.Error())
+		return nil, err
 	}
-	dayDiff := os.Args[1]
-	projectId := os.Args[2]
-	fmt.Println("Removing unused snippets is running...")
-	startDate := time.Now()
 
-	diff, err := strconv.Atoi(dayDiff)
+	return db, nil
+}
+
+func cleanup(dayDiff int, projectId string) error {
+	ctx := context.Background()
+	db, err := createDatastoreClient(ctx, projectId)
 	if err != nil {
-		fmt.Printf("Couldn't convert days to integer from the input parameter, err: %s \n", err.Error())
-		return
+		return err
 	}
 
+	err = db.DeleteUnusedSnippets(ctx, int32(dayDiff))
+	if err != nil {
+		fmt.Printf("Couldn't delete unused code snippets, err: %s \n", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func remove(snippetId string, projectId string) error {
 	ctx := context.Background()
-	pcMapper := mapper.NewPrecompiledObjectMapper()
-	db, err := datastore.New(ctx, pcMapper, projectId)
+	db, err := createDatastoreClient(ctx, projectId)
 	if err != nil {
-		fmt.Printf("Couldn't create the database client, err: %s \n", err.Error())
-		return
+		return err
 	}
 
-	err = db.DeleteUnusedSnippets(ctx, int32(diff))
+	err = db.DeleteSnippet(ctx, snippetId)
 	if err != nil {
-		fmt.Printf("Couldn't delete unused code snippets, err: %s \n", err.Error())
-		return
+		fmt.Printf("Couldn't delete code snippet, err: %s \n", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("expected 'cleanup' or 'remove' subcommands")
+		os.Exit(1)
+	}
+
+	fmt.Println("Removing unused snippets is running...")
```

Comment:
Please consider moving this below the place where we know if this is a cleanup or remove